mlrun/datastore/base.py

```diff
@@ -256,10 +256,14 @@ def reader(*args, **kwargs):
                                 filename = filename.split("/")[-1]
                                 filenames.append(filename)
                         dfs = []
+                        kwargs.pop("filesystem", None)
+                        kwargs.pop("storage_options", None)
```

Comment:
Using fhandle will not work on dask,
Therefore my test in dbfs `test_check_read_df_dir_dd` failed.
Fix it by:
```suggestion
                        if df_module is pd:
                            kwargs.pop("filesystem", None)
                            kwargs.pop("storage_options", None)
                            for filename in filenames:
                                fullpath = f"{base_path}/{filename}"
                                with file_system.open(fullpath) as fhandle:
                                    updated_args = [fhandle]
                                    updated_args.extend(args[1:])
                                    dfs.append(df_module.read_csv(*updated_args, **kwargs))
                        else:
                            for filename in filenames:
                                updated_args = [f"{base_path}/{filename}"]
                                updated_args.extend(args[1:])
                                dfs.append(df_module.read_csv(*updated_args, **kwargs))
```